index.html

```diff
@@ -606,18 +682,130 @@
 	if (slotIndex !== -1) slotComment.value = slotData[slotIndex].comment;
 }
 
+
+// Makes a button on the interface correspoding to the saved slot. 
+// Gives functionalities to the button. 
+// Buttons id is stored as 'key' of the map and the button itself is stored as 'value' 
 let saveSlot = () => {
-	slotData.push({
+	var data = {
 		cFromET: (700-1200*Math.log2(3/2)) + cAbsoluteN.value,
 		eFromET: (700-400-1200*Math.log2(6/5)) + eAbsoluteN.value,
 		fifthDeviation: -cAbsoluteN.value,
 		thirdDeviation: eAbsoluteN.value,
 		comment: slotComment.value
-	});
+	}
+
+	slotData.push(data);
+
+	// figure out whether the column(slot type) specified is valid and then add the button to the interface
+	 if(slotComment.value == "good" || slotComment.value == "medium" || slotComment.value == "bad"){ 	
+		addButtonToInterface(slotComment.value, data);
+	 }
+	
+	
 	slotIndex = slotData.length - 1;
 	updateSlotStatus();
 }
 
+// 'type' specifies the column 
+// 'data' represents the tuning data    
+let addButtonToInterface = (type, data) => {
+
+		// get the index corresponding to the column
+		var slotTypeIndex = slotTypes.get(type);
+
+		// find a 'free' row to place this button
+		var insertIndex = indexOfEmptyRow(slotTypeIndex);
+
+		// construct the button id in with right format 
+		// format : "row(space)column".
+	 	var buttonId = insertIndex + " " + slotTypeIndex;
+
+	 	// make a button in the appropriate place in the table
+		preferenceTable.rows[insertIndex].cells[slotTypeIndex].innerHTML = "<button id = " + "\"" + buttonId +  "\"" + "> " + "</button>";
+
+
+		TuningMap.set(buttonId, data);
+
+		// store the button in consideration
+		var button = document.getElementById(buttonId);
+
+
+		button.addEventListener("click", function(){
+			// change the tuning of the HK to the tuning corresponding to the button
+			// please have a look at this
+			propagateSlotPitches2(this.id);
+
+			// if nothing has been selected, select this button
+			if(slotCurrentlySelected == null){
+				slotCurrentlySelected = this;
+			}
+
+			// change color of button
+			this.classList.add("slot");
+			
+			
+
+			// some button color changing and sound playing logic. 
+			if(this.id == slotCurrentlySelected.id){
+				cMajorHandler();
+			}else{
+				if(isSoundPlaying){
+					cMajorHandler();
+				}
+			}
+
+			slotCurrentlySelected = this;
+
+			// some more button color changing logic
+			nullifySlotColors();	
+		});
+}
+
+// returns a row where a button of type 'column' can be placed.
+let indexOfEmptyRow = (column) => {
+	for(var i = 1; i < numOfTableRows; i++){
+		var row = preferenceTable.rows[i];
+		var cell = row.cells[column];
+		if(cell.firstChild == null || cell.firstChild.tagName != "BUTTON"){
+			return i;
+		}
+	}
+	return "full";		
+}
```

Comment:
Instead of going by columns, have you considered going by row? E.g., Having three rows for Good, Medium, Bad respectively, and appending a button to each row accordingly.